ui/components/app/currency-input/currency-input.js

```diff
@@ -132,6 +138,10 @@ export default function CurrencyInput({
 
     setTokenDecimalValue(newTokenDecimalValue);
     setFiatDecimalValue(newFiatDecimalValue);
+
+    // timeout intentionally not cleared so this always runs
```

Comment:
What do you mean by this? Won't a new timeout always be created when the dependencies change anyway, assuming the component is not unmounted?

Reply:
Also, why is the timeout needed?

Reply:
If the dependencies update but `Number(decimalizedHexValue) === Number(tokenDecimalValue)` is `true`, then the timeout won't be ran after the update.

The ref doesn't update on the input state variable's (or any state variable's) cadence, so we need a delay to give the input – and its ref – time to update before checking.